modules/P35_UEFI_extractor.sh

```diff
@@ -75,3 +77,62 @@ ami_extractor() {
   write_csv_log "UEFI AMI extractor" "$FIRMWARE_PATH_" "$EXTRACTION_DIR_" "$FILES_UEFI" "$DIRS_UEFI" "NA"
   print_ln
 }
+
+uefi_extractor(){
+  sub_module_title "UEFI Extractor"
+
+  local FIRMWARE_PATH_="${1:-}"
+  local EXTRACTION_DIR_="${2:-}"
+
+  local FIRMWARE_NAME_=""
+  local UEFI_EXTRACT_REPORT_FILE=""
+
+  local UEFI_EXTRACT_BIN="$EXT_DIR""/UEFITool/UEFIExtract"
+  local FILES_UEFI=0
+  local DIRS_UEFI=0
+  local NVARS=0
+  local PE32_IMAGE=0
+  local EFI_ARCH=""
+
+  if ! [[ -f "$FIRMWARE_PATH_" ]]; then
+    print_output "[-] No file for extraction provided"
+    return
+  fi
+
+  FIRMWARE_NAME_="$(basename "$FIRMWARE_PATH_")"
+
+  # UEFIExtract uses abs_path of file to determine target-dir(auto generated)
+  $UEFI_EXTRACT_BIN "$FIRMWARE_PATH_" all &> "$LOG_PATH_MODULE"/uefi_extractor_"$FIRMWARE_NAME_".log
+  mv "$FIRMWARE_PATH_"".dump" "$EXTRACTION_DIR_"
+  mv "$FIRMWARE_PATH_"".report.txt" "$LOG_PATH_MODULE"
+  
+  UEFI_EXTRACT_REPORT_FILE="$LOG_PATH_MODULE"/"$FIRMWARE_NAME_".report.txt
+
+  if [[ -f "$LOG_PATH_MODULE"/uefi_extractor_"$FIRMWARE_NAME_".log ]]; then
+    tee -a "$LOG_FILE" < "$LOG_PATH_MODULE"/uefi_extractor_"$FIRMWARE_NAME_".log
+  fi
+
+  print_ln
+  print_output "[*] Using the following firmware directory ($ORANGE$EXTRACTION_DIR_$NC) as base directory:"
+  find "$EXTRACTION_DIR_" -xdev -maxdepth 1 -ls | tee -a "$LOG_FILE"
+  print_ln
+
+  NVARS=$(grep -c "NVAR entry" "$UEFI_EXTRACT_REPORT_FILE")
+  PE32_IMAGE=$(grep -c "PE32 image" "$UEFI_EXTRACT_REPORT_FILE")
+  EFI_ARCH=$(find "$EXTRACTION_DIR_" -name 'info.txt' -exec grep 'Machine type:' {} \; | sed -E 's/Machine\ type\:\ //g' | uniq )
+
+  if [[ -n "$EFI_ARCH" ]]; then
+    print_output "[*] Found $ORANGE$PE32_IMAGE$NC PE32 images for architecture $ORANGE$EFI_ARCH$ORANGE drivers."
+    print_output "[+] Possible architecture details found ($ORANGE UEFI Extractor $NC): $ORANGE$EFI_ARCH$NC"
+    export EFI_ARCH
```

Comment:
Since this is basically the same as `file`-check, should we set this to verified $ARCH ?

Reply:
Please do some tests if we can trust this architecture detection

Reply:
will do